### PR TITLE
More nofollows + New Middleware to set X-Robots-Tag HTTP Header

### PR DESCRIPTION
**Pull request by mlissner**
This PR represents a belt, suspenders, and underwear approach to keeping ~your body covered up~ **pages off Google** while being ~chaste~ **thrifty with our crawl budget**:

 - The **belt** is the nofollow link and the de-listing of these pages from our sitemap.xml file. Hopefully Google never loads these pages in the first place. But if the belt fails...
 - The **suspenders** is the new x-robots-tag HTTP header that this PR lands via a new middleware. And if the suspenders fail...
 - The **underwear** is the use of the noindex meta HTML tag, which is put in place by the `private=True` context variable (as always).

Each of these should help keep pages out of Google. `nofollow` prevents Google from crawling pages. X-Robots-Tag provides an HTTP header to Google to say a page shouldn't be indexed, and does that *before* Google downloads a page, making it more thrifty than the HTML tag. Some search engines don't know what to do with the X-Robots-Tag HTTP header, so the HTML meta tag is a fallback.

That's a lot of theory. In practice what this does is:

1. There's a new middleware that adds the `X-Robots-Tag` HTTP header to responses if the `private` context variable is set to `True`. This middleware only works when it is given TemplateResponses, so this also replaces a bunch of `render` calls with that. Making this replacement allows the middleware to access the context variables.

2. A lot more links now have `nofollow`. 

In addition:

1. It was dumb that our templates were named `view_XYZ`, so I renamed a few of them.

3. Our other middleware was using an old obsolete style, so I upgraded and simplified it. We could probably yank this middleware, but it's fine for now.

3. I slipped in a tweak to a manage.py script you wrote to fix its memory usage.

### CODE REVIEW

File: cl/lib/middleware.py [hunk 1]
```diff
@@ -29,25 +32,59 @@ class MaintenanceModeMiddleware(MiddlewareMixin):
...
         )
         add_never_cache_headers(r)
         return r
+
+
+class RobotsHeaderMiddleware:
+    """Adds x-robots-tag HTTP header to any request that has `private=True`
+
+    There's some evidence and good logic to support the idea that using the
+    x-robots-tag HTTP header uses less of of a site's "crawl budget" than
```
Comment (albertisfu) [inline cl/lib/middleware.py:60]:
```suggestion
    x-robots-tag HTTP header uses less of a site's "crawl budget" than
```

**Review by albertisfu — COMMENTED** [review]
This seems good, the new middleware works well!

I just have a question, I saw that in order for the middleware works is needed to use `TemplateResponse`. So in order to decide where to use it, what would be the criteria?

I saw that it was added to views that don't require login like (simple_pages) and also to views that require login like (users).

For example, I'm not sure if it should be added to some views on alerts like `disable_alert`, `enable_alert`, or `toggle_docket_alert_confirmation` that are public pages but their URL are only shown within an email so that unless someone indexed that email (unlikely?) it shouldn't be necessary to add the header right?

**Comment by mlissner** [comment]
> I just have a question, I saw that in order for the middleware works is needed to use TemplateResponse. So in order to decide where to use it, what would be the criteria?

Yeah, I didn't have a good answer for this, but the middleware only works with the TemplateResponse. If you give it a rendered page, it'll just not do anything (which means we'll fall back to the HTML meta tag — not ideal, but not horrible either). What I was thinking was that if the view is private sometimes, (`private=True`), then we should use the TemplateResponse, and if the other views in the views.py file were private, might as well use the TemplateResponse throughout. It wouldn't be horrible to just always use a TemplateResponse, but I didn't go so far as upgrading all the code to do so.

> For example, I'm not sure if it should be added to some views on alerts like disable_alert, enable_alert, or toggle_docket_alert_confirmation that are public pages but their URL are only shown within an email so that unless someone indexed that email (unlikely?) it shouldn't be necessary to add the header right?

Yeah, I think I looked briefly at these, but since they're not in our sitemaps and not linked from CourtListener pages, it didn't seem to matter too much. But it wouldn't hurt to make them use TemplateResponse. We should get in the habit of doing so, but I was too lazy to upgrade everything.

**Comment by mlissner** [comment]
I don't know what's wrong with mypy, but I'm going to merge this guy. Thanks @albertisfu.

**Pull request merged**